api/agentpb/query.go

```diff
@@ -232,11 +223,7 @@ func makeInterface(value *QueryActionValue) (interface{}, error) {
 		// See https://jira.percona.com/browse/SAAS-107.
 		return string(v.Bytes), nil
 	case *QueryActionValue_Timestamp:
-		t, err := ptypes.Timestamp(v.Timestamp)
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to handle timestamp")
-		}
-		return t, nil
+		return v.Timestamp.AsTime(), nil
```

Comment:
Should probably call [`v.CheckValid()`](https://github.com/protocolbuffers/protobuf-go/blob/v1.27.1/types/known/timestamppb/timestamp.pb.go#L220) to handle any possible resulting error

Reply:
Yep, nice catch! Fixed.